### ISSUE TITLE
Scythe Upgrade

### ISSUE THREAD

**Issue by lstebner** [jeremyckahn/farmhand#166]
add the ability to upgrade the scythe using the smelter. upgrading the scythe will increase the yield of crops. this is blocked by #163, and needs a few more details to really be implemented

**Comment by lstebner** [comment]
@jeremyckahn have you thought about how this could be implemented? i took a look and don't see a very obvious way. it seems like tools are not unique to anyone's game save, so we might have to add something new to game state for tool level tracking. i also noticed tools are not included in the typical "items" maps, so they can't easily be referenced in something like a recipe. and then one other difference i have thought of so far is that these item upgrades only need crafted once, vs being infinitely craftable once unlocked.

this is looking like a big undertaking at the moment; maybe a whole feature of its own (we could include upgrading other tools in that feature too). hopefully you have an idea in mind as to how we can do this 😄

**Comment by jeremyckahn** [comment]
>it seems like tools are not unique to anyone's game save, so we might have to add something new to game state for tool level tracking.

Yeah, we'll need to add some persisted game state to track tool upgrades. I'm thinking `state.toolUpgrades = {}`, where each key is a tool ID (probably a new enum) and the value is the number level that it's been upgraded to.

>tools are not included in the typical "items" maps, so they can't easily be referenced in something like a recipe.

I think we're going to need a new mechanism within the code to handle tool upgrades. It could probably work similarly to Field and Cow Pen upgrades, but I'm not sure there's a lot of code from those systems that could be directly reused. I think we can use the data model I described above to track this.

Broadly speaking, I'm envisioning a new component that's similar in nature to [`TierPurchase`](https://github.com/jeremyckahn/farmhand/blob/develop/src/components/TierPurchase/TierPurchase.js) with a similar UX that also incorporates item requirements into the upgrade. I'm on fence with whether we should augment that component to accept item requirements or just build a new component for this. In any case, I don't think we have to start completely from scratch. In my mind I'm seeing this in the "Upgrades" tab of the Shop.

Certainly a sizable feature, but probably not bigger than the shoveling feature implementation, I'd estimate. Let me know if this helps or if you have any more questions! Also feel free to ping me on Discord and I'll respond when I can. 🙂

**Comment by lstebner** [comment]
hey we did this! :tada:
